content/usgs-packages/Application_Part4.Rmd

```diff
@@ -0,0 +1,268 @@
+
+# download precip data
+precip_stencil <- webgeom(paste0('HUC8::', paste(huc8s, collapse=",")))
+precip_knife <- webprocess() # accept defaults for weighted average
+all_webdata <- query("webdata")
+precip_fabric <- webdata(all_webdata["United States Stage IV Quantitative Precipitation Archive"])
+variables(precip_fabric) <- query(precip_fabric, 'variables')
+times(precip_fabric) <- c(startDate, endDate)
+```
+
+Executing geojob:
+
+```{r execute-job-off, addToggle="ON", eval=FALSE}
+precip_geojob <- geoknife(precip_stencil, precip_fabric, precip_knife)
+wait(precip_geojob, sleep.time = 10) # add `wait` when running scripts
+precip_data <- result(precip_geojob)
+```
+
+```{r read-intermediate, include=FALSE}
+geoknife_list <- readRDS("precip_geoknife.RDS")
+precip_geojob <- geoknife_list$geojob
+precip_data <- geoknife_list$data
+```
+
+Creating plots:
+
+```{r sites-data-plot-continued, addToggle="ON"}
+precip_data_long <- gather(precip_data, huc8, precip, 
+                           -which(!names(precip_data) %in% huc8s))
+
+for(i in sites){
+  huc_site_i <- filter(sb_sites_info, site_no == i)$huc_cd # corresponding HUC8
+
+  precip_site_i <- filter(precip_data_long, huc8 == huc_site_i)
+  nitrogen_site_i <- filter(nitrogen_data, site_no == i)
+  phosphorus_site_i <- filter(phosphorus_data, site_no == i)
+  
+  layout(matrix(1:3, nrow=3))
+  plot(precip_site_i$DateTime, precip_site_i$precip,
+       col="red", pch=20, xlab = "Time", ylab = "Precip accumulation, in",
+       main = paste("Site", i))
+  plot(nitrogen_site_i$sample_dt, nitrogen_site_i$result_va, 
+       col="green", pch=20, xlab = "Time", ylab = "Nitrogren concentration, mg/l")
+  plot(phosphorus_site_i$sample_dt, phosphorus_site_i$result_va,
+       col="blue", pch=20, xlab = "Time", ylab = "Phosphorus concentration, mg/l")
+}
+```
+
+## Create location to publish on SB
+
+The challenge with this application was to provide summaries of precipitation, nitrogen, and phosphorus data for a specific set of sites provided by a cooperator. The challenge was to automate the entire workflow, so the final step is to publish our results to ScienceBase. It would make most since to push the results to the same ScienceBase item that the cooperator provided for sites, but since this is an exercise and others will be completing it, we will save the results to a personal SB item. 
+
+Therefore, the first step is to create a folder under your user to save the results. Title this new item "usgs-pkgs-application-results". Visit the [sbtools modify lesson](/usgs-packages/sbtools-modify) to remind yourself how to do this. Try it on your own before expanding the solution code!
+
+```{r create-new-item, addToggle="ON"}
+# automatically created under the authenticated user
+sb_results_item <- item_create(title = "usgs-pkgs-application-results")
+
+# you would only create the item once, then you could just use its id moving forward
```

Comment:
"you would only create the item once, then you could just use its id moving forward" ->
"only create the item once, then just use its ID moving forward"